File: 1796.py
class Solution:
    def secondHighest(self, s: str) -> int:
        highest, second = -1, -1

        for letter in s:
            if(letter.isnumeric()):
                val = int(letter)
                if(val > highest):
                    highest, second = val, highest
                elif(val != highest and val > second):
                    second = val
        return second

File: 1796.cpp
class Solution {
public:
    int secondHighest(string s) {
        int arr[10];
        bool first = false;

        for(int i = 0; i < s.size(); i++) {
            if(s[i] >= '0' && s[i] <= '9') {
                arr[s[i] - '0']++;
            }
        }
        for(int i = 9; i >= 0; i--) {
            if(arr[i] > 0) {
                if(!first) {
                    first = true;
                } else {
                    return i;
                }
            }
        }
        return -1;
    }
};